File: src/modulo-01-java-OO/Modulo01/ElfoNoturno.java


public class ElfoNoturno extends Elfo
{
    public ElfoNoturno(String nome){
        super(nome);
    }
    public ElfoNoturno(String nome,int flechas){
        super(nome,flechas);
    }
    public void atirarFlecha(Dwarf dwarf) {
        if(super.getFlechas()!=0){
            super.atirarFlecha(dwarf);
            experiencia+=2;
            vida-=(vida/100)*5;
            if(super.vida < 1){
                super.vida=0;
                super.status=Status.MORTO;
            }
        }
    }
}

File: src/modulo-03-web/dia-04/ElfoNoturno.js
function ElfoNoturno() {
  //super(nome,flechas)
  //Elfo.call(this,nome,flechas);
  Elfo.apply(this,arguments);
}

// extends
ElfoNoturno.prototype=Object.create(Elfo.prototype);

//override

//static
ElfoNoturno.alturaMedia=function () {
  return 2.15;
}
